StortSpelProjekt/Game/src/Gamefiles/Components/TeleportComponent.cpp

```diff
@@ -38,6 +38,7 @@ void component::TeleportComponent::OnCollision(Collision* collisionEvent)
 		else
 		{
 			Player::GetInstance().IsInShop(false);
+			EventBus::GetInstance().Publish(&RoundStart());
```

Comment:
Jacob told me a few weeks back that its usually bad practise to publish events inside events, did you think this structure through before you did this?

Reply:
The only other option I see to this is to add an update to the teleportcomponent that checks if a boolean is true or not to call the roundstart event. For that reason I don't see another solution to this problem. I would like to know why this is bad practice though?